server/adaptors/zeppelin_backend.ts

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { NotebookAdaptor } from './notebook_adaptor';
+import { RequestHandlerContext } from '../../../../src/core/server';
+import { optionsType } from '../../common';
+import { requestor } from '../helpers/wreck_requests';
+
+export class ZeppelinBackend implements NotebookAdaptor {
+  backend = 'Zeppelin Backend';
+
+  // Gets all the notebooks available from Zeppelin Server
+  // ZS Endpoint => http://[zeppelin-server]:[zeppelin-port]/api/notebook
+  viewNotes = async function (_context: RequestHandlerContext, wreckOptions: optionsType) {
+    try {
+      let output = [];
+      const body = await requestor('GET', 'api/notebook/', wreckOptions);
+      output = JSON.parse(body.toString()).body;
+      return output;
+    } catch (error) {
+      throw new Error('View Notebooks Error');
```

Comment:
np: print the error message as well- `throw new Error('View Notebooks Error:', error)`

Reply:
+1

Reply:
Changed  to `throw new Error('View Notebooks Error:' + error)`